Sources/Decoder.swift

```diff
@@ -157,6 +157,9 @@ public struct Decoder {
                 
                 for subJSON in jsonArray {
                     guard let model = T(json: subJSON) else {
+                        if T.self is ArrayPartiallyDecodable.Type {
```

Comment:
it's not that I'm deadset against this feature - i just don't want to include it if there's an aspect of confusion to it for end users. if it works in certain cases and not others, Decoding may become unpredictable. 

sorry it didn't work out with this implementation - i do appreciate your time and effort on it!